server/src/utils/ErrorInfo.h

```diff
@@ -0,0 +1,27 @@
+#ifndef UNITTESTBOT_ERRORINFO_H
+#define UNITTESTBOT_ERRORINFO_H
+#include <string>
+#include <optional>
+#include "path/FileSystemPath.h"
+
+enum ErrorType {
+    NO_ERROR,
+    ASSERTION_FAILURE,
+    EXCEPTION_THROWN
+};
```

Comment:
Could you please add some docs about error mode? Probably, comments or doc at wiki?